### PR TITLE
Add documentation to show how to correctly enable user impersonation

### PR DESCRIPTION
**Pull request by frak**
I found issue #174 when trying to enable user impersonation for my own Sonata application, and decided to write the docs that should allow the issue to finally be closed.  This is my first time with RST, but the build seemed to work.  I have done as much proof-reading as possible, and corrections gratefully received.

### CODE REVIEW

File: Resources/doc/reference/user_impersonation.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
+.. index::
+    single: User Impersonation
+
+User Impersonation
+==================
+
+To enable the user impersonation links you must first enable the impersonation feature in your
+firewall. Once you have enabled the feature, you will need to ensure that the user that you wish
+to role switch from has the ``ROLE_SWITCH_ALLOWED`` role.
```
Comment (greg0ire) [inline Resources/doc/reference/user_impersonation.rst]: Really? Or is it `Role_ALLOWED_TO_SWITCH` ?

File: Resources/doc/reference/user_impersonation.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        your_firewall_name:
+            ...
+            switch_user: true
+
+Please note that sessions are not shared by default over different firewalls, to enable this
+feature please refer to the
+`Symfony Documentation <http://symfony.com/doc/current/reference/configuration/security.html#reference-security-firewall-context>`_.
+
+
+Then is it simply a case of providing the URL to redirect to one the user has been impersonated
```
Comment (greg0ire) [inline Resources/doc/reference/user_impersonation.rst]: I don't understand this sentence.

**Comment by frak** [comment]
Hopefully this makes more sense, and thanks for the proof read :)

File: Resources/doc/reference/user_impersonation.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            ...
+            switch_user: true
+
+Please note that sessions are not shared by default over different firewalls (for example, if you
+are trying to switch to a user who is authenticated on another firewall, such as ``main`` given in
+the example above).  To enable this feature please refer to the
+`Symfony Documentation <http://symfony.com/doc/current/reference/configuration/security.html#reference-security-firewall-context>`_.
+
+
+Then is it simply a case of providing the route name to redirect to one the user has been impersonated
```
Comment (greg0ire) [inline Resources/doc/reference/user_impersonation.rst]: Oh I finally get it! You mean "once", not "one".

**Comment by frak** [comment]
@greg0ire Good catch, I didn't even see it when you pointed it out to me, fixed now.

**Comment by greg0ire** [comment]
LGTM @OskarStark @core23

**Comment by OskarStark** [comment]
Good job @frak 👍  !

Thank you! 🎉

**Comment by frak** [comment]
Excellent, I hope this means that #174 can be closed now :)

**Pull request merged**